source/Calamari/Integration/Nginx/NginxServer.cs

```diff
@@ -141,9 +146,16 @@ public NginxServer WithAdditionalLocations(IEnumerable<Location> locations)
         private string SanitizeLocationName(string locationPath, string defaultValue)
         {
             var match = Regex.Match(locationPath, "[a-zA-Z0-9/]+");
-            return match.Success 
-                ? match.Value.Replace("/", "_").Trim('_') 
-                : defaultValue;
+            if (match.Success)
+            {
+                // Watch out for cases where multiple locations both match a string like "/". Both
+                // of these will be sanitized down to an empty string.
+                var sanitizedResult = match.Value.Replace("/", "_").Trim('_');
+                // If we did not get an empty string, use the result. Otherwise fall back to the default.
+                if (!String.IsNullOrWhiteSpace(sanitizedResult)) return sanitizedResult;
```

Comment:
minor nitpick, but this should use `string` rather than `String`...See [this blog post](https://blog.paranoidcoding.com/2019/04/08/string-vs-String-is-not-about-style.html) for additional context as to why

```suggestion
                if (!string.IsNullOrWhiteSpace(sanitizedResult)) return sanitizedResult;
```

Reply:
👍 I'll make this change.